File: client/lib/models/responses/auth_response.dart
import '../user_response.dart';

/// Payload returned by `POST /api/auth/login`, `/oauth/{provider}`, and
/// `/token/refresh`. Mirrors backend `AuthResponse`.
class AuthResponse {
  AuthResponse({
    required this.accessToken,
    required this.refreshToken,
    required this.accessTokenExpiresIn,
    required this.tokenType,
    required this.user,
  });

  final String accessToken;
  final String refreshToken;
  final int accessTokenExpiresIn;
  final String tokenType;
  final UserResponse user;

  factory AuthResponse.fromJson(Map<String, dynamic> json) {
    return AuthResponse(
      accessToken: json['accessToken'] as String,
      refreshToken: json['refreshToken'] as String,
      accessTokenExpiresIn: (json['accessTokenExpiresIn'] as num).toInt(),
      tokenType: json['tokenType'] as String? ?? 'Bearer',
      user: UserResponse.fromJson(json['user'] as Map<String, dynamic>),
    );
  }
}

File: api/src/main/java/com/example/athletehub/dto/AuthResponse.java
package com.example.athletehub.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/** Payload returned by successful login (and later by the refresh endpoint). */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AuthResponse {

    private String accessToken;
    private String refreshToken;
    /** Access token lifetime in ms — the client uses this to schedule refresh before expiry. */
    private Long accessTokenExpiresIn;
    @Builder.Default
    private String tokenType = "Bearer";
    private UserDto user;
}
